misc/synchronization.md

```diff
@@ -325,35 +337,63 @@ Figure 1 shows how the mechanism works, while Figure 2 contains the legend of th
+while it sets a custom exception when it is called after an exception has been thrown.
+In the latter case, the call to `get()` in the continuation will rethrow the exception,
+the promise will be destructed and the new future stored in the matrix will have an invalid shared state.
+Therefore the continuations in `operator()` and `read()` have to handle it.
 The tile destructor sets the promise value if it was called during normal execution, while it sets
 an exception when it is called after an exception has been thrown.
 If the future contains an exception the call to `get()` in the continuation will rethrow the exception,
 the promise will be destructed and the new future stored in the matrix will have an invalid shared state.
 Therefore the continuations in `operator()` and `read()` have to handle this case.
 
-Constant matrix views can be handled in a simple way. On construction it is only needed to call the `read()`
-method of the parent matrix to get a copy of the shared future that is saved in the view and returned
-each time the view is used. To release the view the `done()` method just clears the view shared future.
-
-For non constant matrix view the mechanism works in this way. On construction the futures of the
-parent matrix are moved in the view and replaced by futures of the promises (two for each tile) also
-stored in the view.
-The `operator()` and `read()` methods works in the same way as for the matrix.
-When `doneWrite()` is called the view creates its shared future (if not existent)
-and duplicates the tile included (the promise cannot be dublicated,
-therefore in the copy it is replaced by a new promise (the blue promise in Fig 3 and 4 moved))
-to set the shared future (purple and red arrow pointing to the light yellow promise in Fig 3 and 4).
-When both the blue and purple futures are ready, it means that the tiles of both the two shared futures
-went out of scope, and therefore the reading of the elements of the tiles are completed.
-At this point the main future in the parent matrix can be set. A synchrounous dataflow is used to set
-the light green future value.
-However a copy of the shared future is still included in the matrix view
-(allowing read-only operations to still be scheduled.) The shared future is cleared by the call to
-`done()` after which the tile control is fully back to the parent matrix.
-
-It should be noted that the  `done()` method invokes `doneWrite()` if the tile is still in read-write mode, i.e. when `doneWrite()` hasn't be called yet.
-Figure 3 and 4 show how a non-constant matrix view works.
+### Matrix View
+
+`MatrixView<const T>` can be handled in a simple way. On construction it only needs to call the `read()`
+method of the parent matrix to get a copy of the shared future, which is saved in the view and will be returned
+each time the view is used (`read()`). To release the view the `done()` method just clears the view shared future.
```

Comment:
```suggestion
method of the parent matrix to get a copy of the shared future, which is saved internally and it will be returned
each time it is used (`read()`). To release the view, the `done()` method just clears the view shared future.
```